woocommerce/woocommerce
# CYS: Prevent readding the same pattern

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

* Renames `use-insert-pattern-by-name` hook to `use-pattern-by-name` since we now export a get pattern method from here.
* Refactor existing usage based on the above
* Update onPatternClick handler to remove the pattern if its already present
* Update `is-selected` class utility function to also `.blur()` the element
* Add a removePattern method to the `use-insert-pattern` hook

Closes https://github.com/woocommerce/woocommerce/issues/50388



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Load CYS experience
2. Click Design Your Homepage
3. Add a pattern and check its inserted into the preview
4. Click the same pattern to remove it, check it is also removed from preview.
5. Check that the focus() state does not remain on the pattern thumbnail (dont confuse with hover state)



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
